src/query/functions/utils/heap_test.go

```diff
@@ -216,3 +220,133 @@ func TestNegativeCapacityHeap(t *testing.T) {
 		assert.Equal(t, testArray[pair.Index], pair.Val)
 	}
 }
+
+func equalPairs(t *testing.T, expected, actual []ValueIndexPair) {
+	e := fmt.Sprint(expected)
+	a := fmt.Sprint(actual)
+	assert.Equal(t, e, a)
```

Comment:
Matching strings here seems odd; can this not use `EqualsWithNaNs`?

Reply:
It doesn't work when using `EqualsWithNaNs`:
```
panic: interface conversion: interface {} is []utils.ValueIndexPair, not string [recovered]
	panic: interface conversion: interface {} is []utils.ValueIndexPair, not string
```

Reply:
You would need to either extend EqualsWithNaNs to take this type, or better to compare the values by hand

Reply:
https://github.com/m3db/m3/pull/2792/commits/bd3f326bd21383bca9c81fc1f51927a935bf619f#diff-cf04475fadbfc80ae239ceb87d393ce927af7b017dd971ad6b741cee31e15dcbR224-R228